lib/backend.js

```diff
@@ -9,16 +9,52 @@ const pickBy = require('lodash/pickBy')
 const omit = require('lodash/omit')
 const querystring = require('querystring')
 const clc = require('cli-color')
-
 require('es7-object-polyfill')
 
+process.env.AWS_SDK_LOAD_CONFIG=1
+const AWS = require('aws-sdk')
+
+/**
+ * Create the configuration for the local dynamodb instance.
+ *
+ * Region and AccessKeyId are determined as follows:
+ *   1) Look at local aws configuration in ~/.aws/credentials
+ *   2) Look at env variables env.AWS_REGION and env.AWS_ACCESS_KEY_ID
+ *   3) Use default values 'us-east-1' and 'key'
+ *
+ * @param env - the process environment
+ * @returns {{endpoint: string, sslEnabled: boolean, region: string, accessKeyId: string}}
+ */
+function loadDynamoConfig(env) {
+  const dynamoConfig = {
+    endpoint: 'http://localhost:8000',
+    sslEnabled: false,
+    region: 'us-east-1',
+    accessKeyId: 'key'
+  }
+
+  if (AWS.config) {
+    dynamoConfig.region = AWS.config.region
+
+    if (AWS.config.credentials) {
+      dynamoConfig.accessKeyId = AWS.config.credentials.accessKeyId
+    }
+  }
+
+  if (env.AWS_REGION) {
+    dynamoConfig.region = env.AWS_REGION
+  }
+
+  if (env.AWS_ACCESS_KEY_ID) {
+    dynamoConfig.accessKeyId = env.AWS_ACCESS_KEY_ID
+  }
+
+  return dynamoConfig
+}
+
 const createAwsConfig = () => {
   const env = process.env
-  const awsConfig = {
-    region: env.AWS_REGION || 'us-east-1',
-    accessKeyId: env.AWS_ACCESS_KEY_ID || 'key', // DynamoDB Local doesn't care what the key/secret are
-    secretAccessKey: env.AWS_SECRET_ACCESS_KEY || 'secret'
-  }
+  const dynamoConfig = loadDynamoConfig(env)
```

Comment:
Could you move config reading after the if/else block below? There is no point in trying to read config if we fail the amazonaws.com check below.